File: app/src/main/java/com/riders/thelab/data/remote/api/ArtistsAPIService.java
package com.riders.thelab.data.remote.api;

import com.riders.thelab.data.remote.dto.artist.Artist;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Url;

public interface ArtistsAPIService {

    @GET
    Single<List<Artist>> getArtists(@Url String url);

    @GET("{artistsPath}")
    Single<List<Artist>> getArtistsWithPath(@Path(value = "artistsPath") String path);
}

File: app/src/main/java/com/riders/thelab/data/remote/api/ArtistsAPIService.kt
package com.riders.thelab.data.remote.api

import com.riders.thelab.data.remote.dto.artist.Artist
import io.reactivex.rxjava3.core.Single

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Url

interface ArtistsAPIService {
    @GET
    fun getArtists(@Url url: String): Single<List<Artist>>

    @GET("{artistsPath}")
    fun getArtistsWithPath(@Path(value = "artistsPath") path: String?): Single<List<Artist>>
}